src/components/IconChooser.js

```diff
@@ -6,8 +6,12 @@ import AppStyles from '../styles'
 
 import AccountIcon from './AccountIcon'
 
-import { words } from '../util/random'
-import { brainWalletAddress } from '../util/native'
+import { brainWalletAddress, words } from '../util/native'
+
+async function refreshIcons (self) {
```

Comment:
Why do we move it outside of the `Component`? You also need to try-catch asynchrounous `setState` invocations.
I think:
```js
refreshIcons = async () => {
  const icons = await ...;
 
  // try is required, since the component might not be mounted any more.
  try {
   this.setState({ icons });
  } catch (e) {
     console.error(e);
  }
}
```
Should work as well.